### ISSUE TITLE
how to use expression in [visible]

### ISSUE THREAD

**Issue by user140547** [isaacplmann/angular2-contextmenu#16]
According to the documentation, this should work:

`<template contextMenuItem let-item [visible]="item.type === 'type1'" >`

I tried to used that, but I could not get it to work. `item` seems to be undefined in `[visible]` property bindings (in the template itself, it works).
I created a [plunker ](http://plnkr.co/edit/XSUSRtpiCVphRUknbjDq?p=preview) and a [stack overflow question](http://stackoverflow.com/questions/39916382/angular2-ngoutletcontext-in-property-bindings), but I can't get it to work. Is this an issue or am I doing something wrong?

**Comment by isaacplmann** [comment]
This is a bug in the documentation. In my app, I'm actually tracking the currently selected item in the parent component and using that in the template [visible] and [enabled] properties.

Like this:

```js
onContextMenu($event: MouseEvent, item: any): void {
  this.selectedItem = item;
  this.cms.show.next({ event: $event, item: item });
  $event.preventDefault();
}
```

And:

```html
<template contextMenuItem let-item [enabled]="selectedItem?.name == 'John'">
  ..
</template>
```

I'll update the documentation so others aren't lead down this path.

**Comment by Metal-Predator** [comment]
The same situation. Actual problem: [visible] and [enabled] bindings doesn't refresh when item attached.

**Comment by isaacplmann** [comment]
@Metal-Predator I commented seconds before you did.  Does my solution work for you?

**Comment by Metal-Predator** [comment]
Just tested. Yes, after fixing bindings works fine. Thanks!

**Comment by isaacplmann** [comment]
Here's another alternative. I can accept a boolean or a function for the enabled and visible inputs. Functions would then be evaluated with the item value - which would mean you don't have to keep track of the selected item in your component.

Like this:

```js
// parent.component.ts
public canEnable(item: any): boolean {
  return item.type === 'type1';
}
```

```html
<!-- parent.component.html -->
<context-menu>
<template contextMenuItem let-item [enabled]="canEnable">
  ..
</template>
</context-menu>
```

@Metal-Predator, @user140547 what do you think?

**Comment by Metal-Predator** [comment]
@isaacplmann , yes it will be perfect of course if it's possible. Better do not introduce extra context

**Comment by user140547** [comment]
@isaacplmann Yes of course it would be a good idea if it can be implemented that way.

**Comment by isaacplmann** [comment]
Now implemented in version 0.4.0

**Comment by user140547** [comment]
I have tried it now, it works for visible, however, for enabled I still have some problems as illustrated in http://plnkr.co/edit/bj1177ZB13jnxQAj1RrX?p=preview
The context menu's first item should be disabled for Joe, but it still works.
The third item is visible only for John, so this works.

**Comment by isaacplmann** [comment]
Yep, that's a bug.  Working on a fix now.

**Comment by isaacplmann** [comment]
Fixed in 0.4.1
